src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,150 @@
+package duke;
+
+/**
+ * Parser for Duke commands, with access to its task list, and storage.
+ */
+public class Parser {
+    private TaskList tasks;
+    private Storage storage;
+
+    /**
+     * Creates a Parser object for the Duke instance.
+     *
+     * @param tasks   The current TaskList used.
+     * @param storage The current Storage used.
+     */
+    public Parser(TaskList tasks, Storage storage) {
+        this.tasks = tasks;
+        this.storage = storage;
+    }
+
+    /**
+     * Passing the command to the relevant methods.
```

Comment:
It is recommended for the method header to start with 'passes' instead of 'passed'.